### DISCUSSION TITLE
Wallet disconnects when I connect to different networks

### DISCUSSION THREAD

**Discussion by GrahamMcBain** [WalletConnect/walletconnect-monorepo#1638] [Developer Support]
Note : I'm using Vanilla JS with web3modal following this guide https://github.com/WalletConnect/web3modal/blob/V2/docs/html.md

I'm currently trying to implement web3modal v2 to my website and i got it working fine
Except that i wanted to add more supported chains ( In the ClientCtrl.setEthereumClient config ) 

When i login with Eth or Goerli, everything is fine
However, when i login with BSC or Polygon , i disconnects 1 second after connecting

This bug only happens when the popup ask to change network at the login ( So if for example, you are on eth Mainnet, and click on BSC then login , it will disconnect you 1 second after ) 

( I added some code to show the problem ) 

Have anyone an idea ? Or has anyone the same problem ?

![code-biggie](https://user-images.githubusercontent.com/36456059/201790960-f1b6fbcf-b0b4-443d-a0e5-631ac1f6db1f.png)
![FheM9scXoAEWtND](https://user-images.githubusercontent.com/36456059/201790963-9c05a0cb-e6e5-44de-8f77-5a4abbc1c1e1.jpeg)

**Comment by xzilja** [comment]
Hey, we have released beta.4 which under the hood get's rid of our custom react hooks and instead provides full support for https://wagmi.sh/

Wagmi is a great library that we were already using before, but were not able to provide full integration for until now.

This means that you can / need to now import hooks from their package and thus get a more reliable and stable developer experience.

Please see new documentation for more info https://docs.walletconnect.com/2.0/introduction/web3modal/about

Vanilla Users - There are no more controllers, instead you can use methods exposed by @wagmi/core package. Please see updated example. (We are working with wagmi team to document this core package, meanwhile intelisense / typescript should give you a good idea of what is available). @wagmi/core package is what powers wagmi under the hood, it exposes a bunch of methods similar to their hooks i.e. useAccount becomes getAccount /watchAccount https://github.com/WalletConnect/web3modal/tree/V2/examples/html
